src/main/java/cn/nukkit/entity/item/EntityItem.java

```diff
@@ -180,8 +180,11 @@ public boolean onUpdate(int currentTick) {
         boolean hasUpdate = this.entityBaseTick(tickDiff);
 
         boolean lavaResistant = item != null && item.isLavaResistant();
-        
-        if (!lavaResistant && isInsideOfFire()) {
```

Comment:
The implementation of `Entity#isInsideOfFire` checks if the colliding blocks are an instance of `BlockFire` which includes `BlockFireSoul`.  The removal of it and subsequent usage of a manual local check could lead to harder maintainability, if another fire-like block were to be added in the future.